Source/Core/Core/PowerPC/Jit64/Jit_LoadStore.cpp

```diff
@@ -233,37 +233,22 @@ void Jit64::dcbx(UGeckoInstruction inst)
   JITDISABLE(bJITLoadStoreOff);
 
   X64Reg addr = RSCRATCH;
-  X64Reg value = RSCRATCH2;
   RCOpArg Ra = inst.RA ? gpr.Use(inst.RA, RCMode::Read) : RCOpArg::Imm32(0);
   RCOpArg Rb = gpr.Use(inst.RB, RCMode::Read);
   RCX64Reg tmp = gpr.Scratch();
   RegCache::Realize(Ra, Rb, tmp);
 
   MOV_sum(32, addr, Ra, Rb);
+  AND(32, R(addr), Imm8(~31));
 
-  // Check whether a JIT cache line needs to be invalidated.
-  LEA(32, value, MScaled(addr, SCALE_8, 0));  // addr << 3 (masks the first 3 bits)
-  SHR(32, R(value), Imm8(3 + 5 + 5));         // >> 5 for cache line size, >> 5 for width of bitset
-  MOV(64, R(tmp), ImmPtr(GetBlockCache()->GetBlockBitSet()));
-  MOV(32, R(value), MComplex(tmp, value, SCALE_4, 0));
-  SHR(32, R(addr), Imm8(5));
-  BT(32, R(value), R(addr));
-
-  FixupBranch c = J_CC(CC_C, true);
-  SwitchToFarCode();
-  SetJumpTarget(c);
   BitSet32 registersInUse = CallerSavedRegistersInUse();
   ABI_PushRegistersAndAdjustStack(registersInUse, 0);
```

Comment:
The main performance enhancement of the assembly version instead of calling the C++ handler is to avoid flushing the register. However the push/pop combination is often even worse then just flushing the register cache. So I'd suggest to use a `gpr.Flush(); fpr.Flush();` instead for unconditional C++ calls.

Reply:
Would there be any performance advantage to doing that compared to falling back to the interpreter (which also does a `gpr.Flush(); fpr.Flush();`)?

Reply:
Not really. Once you flush the register cache (or use push/pop), the speedup is vanished.

Reply:
From limited inspection, it looks like these instructions are generally used in tight loops in small functions. In such cases, only a small number of guest registers need to be flushed (or alternatively, host registers saved), so the exact mechanism does not make a big difference to the generated code.

Interpreter fallback involves an extra call to Interpreter::dcb*(), which in turn calls JitInterface::InvalidateICache(), and the effective address calculation will take a few more instructions than the JIT. That said, it does have the benefit of being less code, so if any change is to be made, I am leaning in this direction.

Reply:
> Interpreter fallback involves an extra call to Interpreter::dcb*(), which in turn calls JitInterface::InvalidateICache(), and the effective address calculation will take a few more instructions than the JIT.

Ah, I hadn't considered that. I suppose it is better to go with something like the current solution (either with push/pop or with flush) instead of the interpreter fallback then.

Reply:
In IRC, degasus suggested to leave this as-is.